sklearn/isotonic.py

```diff
@@ -271,6 +272,7 @@ def _build_y(self, X, y, sample_weight, trim_duplicates=True):
         check_consistent_length(X, y, sample_weight)
         X, y = [check_array(x, ensure_2d=False) for x in [X, y]]
 
+        X = as_float_array(X)
```

Comment:
@TomDLT this has different behavior than `as_float_array` is the input is int32 (and maybe others too?)

```python
import numpy as np
from sklearn.utils import as_float_array, check_array

for dtype in (np.int32, np.int64, np.uint32, np.uint64, np.float32, np.float64):
    print('input dtype\t', dtype.__name__)
    x = np.arange(5).astype(dtype)
    x = as_float_array(x)
    print('as_float_array\t', x.dtype)

    x = np.arange(5).astype(dtype)
    x = check_array(x, dtype=[np.float64, np.float32], ensure_2d=False)
    print('check_array\t', x.dtype)
    print()
```

outputs

```
input dtype	 int32
as_float_array	 float32
check_array	 float64

input dtype	 int64
as_float_array	 float64
check_array	 float64

input dtype	 uint32
as_float_array	 float64
check_array	 float64

input dtype	 uint64
as_float_array	 float64
check_array	 float64

input dtype	 float32
as_float_array	 float32
check_array	 float32

input dtype	 float64
as_float_array	 float64
check_array	 float64
```

Has there been a decision in the other 32-bit prs what to do in this situation?